lib/bundler/rubygems_integration.rb

```diff
@@ -67,6 +67,9 @@ def configuration
       Bundler.ui.error "#{e.class}: #{e.message}"
       Bundler.ui.trace e
       raise Gem::SystemExitException
+    rescue Psych::SyntaxError => e
+      raise YAMLSyntaxError.new(e, "Your RubyGems configuration file (usually located at ~/.gemrc) " \
```

Comment:
(Although we should say something like "The error is probably in your main configuration file, located at $FILENAME", since it might also be in one of those other files.)